### PR TITLE
Drop unused criteria field and these reference in components

### LINKED ISSUE #2026: Drop unused criteria field and these reference in components

### PR DESCRIPTION
**Pull request by LamaEats**
## PR includes

- [x] Refactor
- [x] DX

## Related issues

Resolve #2026

### CODE REVIEW

**Review by awinogradov — APPROVED** [review]

File: src/utils/db.ts [hunk 1]
```diff
@@ -480,14 +480,7 @@ export const updateLinkedGoalsByProject = async (projectId: string, activityId:
                 // mark criterias with these goals as deleted
                 const updateCriterias = ctx.goalAchieveCriteria.updateMany({
                     where: {
-                        OR: [
-                            {
-                                criteriaGoalId: { in: goalIds.map(({ id }) => id) },
```
Comment (9teen90nine) [inline src/utils/db.ts:485]: It will cause prod to 500 if it's still used in current release

Comment (LamaEats) [reply src/utils/db.ts:485]: Oh, well needs create one more pr

Comment (LamaEats) [reply src/utils/db.ts:488]: https://github.com/taskany-inc/issues/pull/2135

Comment (LamaEats) [reply src/utils/db.ts:488]: Now, this pr contain change only in trpc procedures

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**